recipe_scrapers/schoolofwok.py

```diff
@@ -0,0 +1,172 @@
+        else:
+            possibleingredients = (
+                self.soup.find("section", {"id": "recipe-ingredients"})
+                .findChildren(
+                    "div",
+                    {
+                        "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                    },
+                )[0]
+                .find_all("li")
+            )
+            ingredientslist = [
+                normalize_string(ingredient.get_text())
+                for ingredient in possibleingredients
+            ]
+
+        return ingredientslist
+
+    def ingredient_groups(self):
+        possibleingredients = (
+            self.soup.find("section", {"id": "recipe-ingredients"})
+            .findChildren(
+                "div",
+                {
+                    "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                },
+            )[0]
+            .find_all("p")
+        )
+        groupings = defaultdict(list)
+        current_heading = None
+
+        if possibleingredients:
+            for ingredient in possibleingredients:
+                normalizedingredient = normalize_string(ingredient.get_text())
+                if normalizedingredient:
+                    if ingredient.find("strong"):
+                        current_heading = normalizedingredient
+                    else:
+                        groupings[current_heading].append(normalizedingredient)
+        else:
+            possibleingredients = (
+                self.soup.find("section", {"id": "recipe-ingredients"})
+                .findChildren(
+                    "div",
+                    {
+                        "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                    },
+                )[0]
+                .find_all()
+            )
+            for ingredient in possibleingredients:
+                normalizedingredient = normalize_string(ingredient.get_text())
+                if normalizedingredient:
+                    if (
+                        ingredient.name == "h3"
+                        and normalizedingredient.lower() != "ingredients"
+                    ):
+                        current_heading = normalizedingredient
+                    elif ingredient.name == "li":
+                        groupings[current_heading].append(normalizedingredient)
+
+        return [
+            IngredientGroup(purpose=heading, ingredients=items)
+            for heading, items in groupings.items()
+        ]
+
+    def instructions(self):
+        instructions = (
+            self.soup.find("section", {"id": "recipe-ingredients"})
+            .findChildren(
+                "div",
+                {
+                    "class": "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
+                },
+            )[1]
+            .find_all("p")
+        )
```

Comment:
```suggestion
        section = self.soup.find("section", {"id": "recipe-ingredients"})
        div_class = "flex flex-col gap-y-4 font-medium md:basis-1/2 xl:basis-2/5"
        divs = section.findChildren("div", {"class": div_class})

        if len(divs) < 2:
            return ""

        instructions = divs[1].find_all("p")
```
Small updates to match the structure of the other functions if you decide to merge these suggestions.